### PR TITLE
Replace pycodestyle with flake8

### PR DESCRIPTION
**Pull request by wxtim**
Other projects related to this one have been moving to using flake8.
This change proposed for the sake of consistency.

### CODE REVIEW

**Review by oliver-sanders — APPROVED** [review]

**Review by MetRonnie — APPROVED** [review]

**Comment by MetRonnie** [comment]
Should we add flake8 to https://github.com/metomi/isodatetime/blob/fbe537d6ecdf10af8aac6fa415a13a8c1df384b4/setup.py#L98

**Comment by oliver-sanders** [comment]
Yes but I let it slip since pyccodestyle wasn't there before.

If adding it there (good idea) add the tests_require to an `all` group so we can `pip install -e .[all]` as with other repos.

**Comment by wxtim** [comment]
I have now modified setup.py to include flake8 and the option to use `pip install -e .[all]`

File: .travis.yml [hunk 1]
```diff
@@ -43,11 +43,11 @@ install:
   # PyTest to use environment variables in its configuration (e.g. TZ=UTC).
   - pip install coverage pytest-cov pytest-env pytest-runner
   # Code style dependencies
-  - pip install pycodestyle
+  - pip install flake8
   - pip install -e .
```
Comment (oliver-sanders) [inline .travis.yml:47]:
```suggestion
  - pip install -e .[all]
```

**Review by MetRonnie — APPROVED** [review]
I learnt a thing or two about how `pip install` works from this 👍

**Comment by oliver-sanders** [comment]
Just waiting for tests to pass...

**Comment by wxtim** [comment]
> I learnt a thing or two about how pip install works from this +1

Moi aussi!

**Pull request merged**